File: Singleton/singleton01.js
class Singleton {
    constructor() {
        if (!Singleton.instance) {
            Singleton.instance = this;
            this.value = 0; 
        }
        return Singleton.instance;
    }

    increment() {
        this.value += 1;
        console.log(`Value: ${this.value}`);
    }
}

const s1 = new Singleton();
const s2 = new Singleton();
const s3 = new Singleton();

console.log(s1 === s2); // true
console.log(s3 === s1); // true

s1.increment(); // Value: 1
s2.increment(); // Value: 2
s3.increment(); // Value: 3

File: Exemplos/SingletonEX/singleton01.py
class Singleton:
    _instance = None  # variável de classe para armazenar a instância

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(Singleton, cls).__new__(cls)
            cls._instance.value = 0  # inicializa o valor
        return cls._instance

    def increment(self):
        self.value += 1
        print(f"Value: {self.value}")


# Testando
s1 = Singleton()
s2 = Singleton()
s3 = Singleton()

print(s1 is s2)  # True
print(s3 is s1)  # True

s1.increment()  # Value: 1
s2.increment()  # Value: 2
s3.increment()  # Value: 3